rfcs/0006-configuration-constants.md

```diff
@@ -0,0 +1,128 @@
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+When a Sway program is compiled, a set of artifacts are produced. Included in these artifacts are a JSON ABI descriptor, a configuration-time-constant descriptor, and the actual bytecode. The object we are discussing here is the configuration-time-constant descriptor, which describes the specific offset within the bytecode, in terms of bytes, to the data section entry for a particular configuration variable.
+
+The configuration variables for a program can be identified by the `configurable` keyword within the Sway language. This denotes a constant within the program that has no known value at compile time. The value will be provided at configuration time by the SDK or some other consumer of the configuration-time-constant descriptor. As an example:
+
+```rust
+// main.sw
+script;
+
+configurable {
+    CONTRACT_ADDRESS: b256,
+    PRICE_RATIO: u64,
+}
+
+fn main() { ... }
+```
+```
+// configurable-constants.json
+
+{
+    "CONTRACT_ADDRESS": {
+        "type": "b256",
+        "offset": "12345"
+    }
+}
+```
+
+If you are not using the SDK, it will be possible to define these values by specifying a `constants.sw` flag. This will work with the [notion of contract dependencies](https://github.com/FuelLabs/sway-rfcs/pull/19#issuecomment-1286351753). This Sway file must export the required constants as specified by the `configurable` block.
+
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+The `type` field of the JSON file should describe the type of the configurable in the same manner as types are described in the JSON ABI file.
+
+No two configurables in a program should have the same name to prevent collisions.
+
+The range from the offset to the offset plus the size of the type will represent bytes within the bytecode that will be overwritten by the SDK, which will write values according to the ABI encoder's memory layout, which is consistent with Sway's memory layout.
+
+The change is not breaking as it is entirely new functionality.
+
+Configuration time constants may not be defined in any library code and must be defined at the top level. Should a library need to reference a configurable value, it should utilize design patterns such as the [builder pattern](https://en.wikipedia.org/wiki/Builder_pattern) to ingest the constants from the top level.
+
+The `pub` keyword will operate as normal: if it is `pub`, then it is public and importable from elsewhere in the Sway program. If not, then it is only referencable in the scope it is defined within.
```

Comment:
So these two are a big conflicting now since `pub` is not really relevant outside a library for the time being. Besides, with the new syntax, `pub` does not really have a place?